src/timepicker/timepicker.component.ts

```diff
@@ -196,101 +183,58 @@ export class TimepickerComponent
     return Math.sign($event.deltaY as number) * -1;
   }
 
-  ngOnChanges(changes: SimpleChanges): void {
+  ngOnChanges(): void {
     this._store.dispatch(
       this._timepickerActions.updateControls(getControlsValue(this))
     );
   }
 
-  changeHours(step: number, source: TimeChangeSource = ''): void {
-    this.resetValidation();
-    this._store.dispatch(this._timepickerActions.changeHours({ step, source }));
-  }
+  changeTimeOption(option: string, step: number, source: TimeChangeSource = ''): void {
+    const action: { [key: string]: string } = {
+      hours: 'changeHours',
+      minutes: 'changeMinutes',
+      seconds: 'changeSeconds'
+    };
 
-  changeMinutes(step: number, source: TimeChangeSource = ''): void {
-    this.resetValidation();
     this._store.dispatch(
-      this._timepickerActions.changeMinutes({ step, source })
+      (this as any)._timepickerActions[action[option]]({ step, source })
     );
   }
 
-  changeSeconds(step: number, source: TimeChangeSource = ''): void {
-    this.resetValidation();
-    this._store.dispatch(
-      this._timepickerActions.changeSeconds({ step, source })
-    );
+  // old api methods
+  changeHours(step: number, source: TimeChangeSource = ''): void {
```

Comment:
Add jsdoc note and add `warnOnce('Timepicker Component method change** will be removed in next major version, use changeTimeOption instead)`
```
/**
@deprecated
*/
```